### PR TITLE
Doctests

### PR DESCRIPTION
**Pull request by videlec**
The branch
- fix the travis script
- converts all Sage doctest to Python doctests
- fix many doctest failures with Python 2.7

See also [trac #22899](https://trac.sagemath.org/ticket/22899) for the doctests that will be directly moved to Sage.

### CODE REVIEW

**Comment by jdemeyer** [comment]
Why did you change the formatting of the doctests (apart from just changing `sage:` to `>>>`)? This is going to look bad if we would process the documentation with Sphinx.

**Comment by videlec** [comment]
The `sage:` to `>>>` is the minimum to have `doctest` works. The `EXAMPLES` and `TESTS` were Sage specific. What is the minimum to have sphinx compatible syntax? (some websites mention `:Example:` and others `.. doctests::`)

**Comment by jdemeyer** [comment]
> What is the minimum to have sphinx compatible syntax?

The double colon as in `EXAMPLES::` (but also `Some other text::` would work), followed by an indented block.

**Comment by videlec** [comment]
>> What is the minimum to have sphinx compatible syntax?

> The double colon as in EXAMPLES:: (but also Some other text:: would work), followed by an indented block.

Easy to fix. I propose to postpone this change after #13 (Python 3 compatibility).

@jdemeyer could you modify the travis script to check that building the documentation with sphinx works?

**Comment by jdemeyer** [comment]
> could you modify the travis script to check that building the documentation with sphinx works?

I'll try to do that this week but I cannot promise anything.

**Comment by jdemeyer** [comment]
Actually, it turns out that Sphinx does understand the docstring style introduced in this PR. So nothing needs to be fixed...

**Comment by videlec** [comment]
very good. Could you have a look at #13 ?

**Comment by jdemeyer** [comment]
> could you modify the travis script to check that building the documentation with sphinx works?

Done! And it actually works.

File: cypari2/gen.pyx [hunk 1]
```diff
@@ -4512,18 +4844,21 @@ cdef Gen list_of_Gens_to_Gen(list s):
...
-        [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
-        sage: objtogen(i**2 for i in range(5))
-        [0, 1, 4, 9, 16]
-        sage: objtogen([pari("Mod(x, x^2+1)")])
-        [Mod(x, x^2 + 1)]
-        sage: objtogen([])
-        []
+    Tests:
+
+    >>> from six.moves import range
```
Comment (defeo) [inline cypari2/gen.pyx]: I just realised this doctests adds a useless dependency on six. Without the import, the doctest works just fine in Py2&3.
However, I'm not sure about the original intent of the test. Maybe replacing `range(10)` with `list(range(10))` is closer to it.
Opinions? I shall start a separate pull request for this.

**Comment by jdemeyer** [comment]
> I just realised this doctests adds a useless dependency on six.

You could do a similar thing as some other Python 2/3 doctests and just use `sys.version` (or whatever it's called). The idea is to get `xrange` in Python 2 and `range` in Python 3.

**Comment by defeo** [comment]
Oh, I see. I had interpreted it the other way round.

So a one line replacement compatible with both 2 & 3 would be `objtogen(i for i in range(10))`. But then this is redundant with the following doctest, we could as well remove it. I have the impression that it would be more interesting to replace it with `objtogen(list(range(10))`.

**Comment by jdemeyer** [comment]
> So a one line replacement compatible with both 2 & 3 would be objtogen(i for i in range(10)). But

Not quite since `(i for i in range(10))` is not the same as neither `xrange(10)` or `range(10)`. Both are iterables yielding the same numbers, but it's not the same thing.

**Pull request merged**